### PR TITLE
docs: add initial naming guidelines

### PR DESCRIPTION
**Pull request by wincent**
This came up in code review again:

https://github.com/liferay/clay/pull/3684#discussion_r484662169

So I figure it is time to finally document this.

For now starting with abbreviations, but we can add more naming-related things in the future. It is one of the two hard problems, after all...

The examples of "bad" abbreviations are all taken from past code reviews.

Not intended to be exhaustive advice, but it's a start.

Closes: https://github.com/liferay/liferay-frontend-guidelines/issues/159

### CODE REVIEW

File: general/naming.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+| :x: `cb`           | :white_check_mark: `callback`               |
+| :x: `desc`         | :white_check_mark: `description`            |
+| :x: `e`/`err`      | :white_check_mark: `error`                  |
+| :x: `el`           | :white_check_mark: `element`                |
+| :x: `evt`          | :white_check_mark: `event`                  |
+| :x: `fm`           | :white_check_mark: `form`                   |
+| :x: `fmt`          | :white_check_mark: `format`                 |
+| :x: `k`/`v`        | :white_check_mark: `key`/`value`            |
+| :x: `idx`          | :white_check_mark: `index`                  |
+| :x: `img`          | :white_check_mark: `img`                    |
```
Comment (julien) [inline general/naming.md:30]: I think you probably meant  `:white_check_mark: image`

Comment (wincent) [reply general/naming.md:30]: Good `catch`! Thanks.

our-first-catch-of-the-day

Comment (wincent) [reply general/naming.md]: Fixed in: https://github.com/liferay/liferay-frontend-guidelines/pull/195/commits/db9dc2125f74f1d74caf44d6144483982ff006ad

Comment (julien) [reply general/naming.md]: :tada:

Comment (julien) [reply general/naming.md]: It's nice to have both **good** and **bad** examples - never would have figured out that `prj` stands for `project`

**Comment by wincent** [comment]
I think this has baked long enough, and I want to link to it, so... merging.

**Pull request merged**